test/ngMock/angular-mocksSpec.js

```diff
@@ -1506,6 +1506,42 @@ describe('ngMock', function() {
     });
 
 
+    it('should throw error when expectation fails', function() {
+      expect(function() {
+        hb.expectPOST('/some', {foo: 1}).respond({});
+        hb('POST', '/some', {foo: 2}, callback);
+        hb.flush();
+      }).toThrowError(/^Expected POST \/some with different data/);
+    });
+
+
+    it('should throw error when expectation about headers fails', function() {
+      expect(function() {
+        hb.expectPOST('/some', {foo: 1}, {X: 'val1'}).respond({});
+        hb('POST', '/some', {foo: 1}, callback, {X: 'val2'});
+        hb.flush();
+      }).toThrowError(/^Expected POST \/some with different headers/);
+    });
+
+
+    it('should throw error when expectation about both data and headers fails', function() {
+      expect(function() {
+        hb.expectPOST('/some', {foo: 1}, {X: 'val1'}).respond({});
+        hb('POST', '/some', {foo: 2}, callback, {X: 'val2'});
+        hb.flush();
+      }).toThrowError(/^Expected POST \/some with different/);
```

Comment:
Different what? :smiley:

Reply:
Is it important? :)

Reply:
It is :smiley:
It ensures that the data error takes precedence. It is not terribly important, but there is no need to break this behavior in the future (as originally happened in this PR).